scripts/aida_m18_submission_checker.py

```diff
@@ -0,0 +1,370 @@
+            if directory in "NIST" directory and no .ttl in "NIST" directory, then 1b   --> .*\/NIST\/\S*\/\S*\.ttl mode =/i
+            
+        '''
+        #type count and ta1_temp used to determine if 1a or 1b
+        type_count = 0
+        ta1_ret_val = 0
+
+        names_str = " ".join(ARCHIVE_NAMES)
+        ta1a_dir_struct_regex = re.compile(".*\/NIST\/[^\/]+.ttl", re.IGNORECASE)
+        ta1b_dir_struct_regex = re.compile(".*\/NIST\/\S*\/\S*\.ttl", re.IGNORECASE)
+
+        #if ttls located in the nist directory, mark as potential TA_1a
+        if ta1a_dir_struct_regex.search(names_str) is not None:
+            ta1_ret_val = 0
+            type_count = type_count + 1
+        #if ttls located at some subdirectory of the nist directory, mark as potential TA_1b
+        if ta1b_dir_struct_regex.search(names_str) is not None:
+            ta1_ret_val = 1
+            type_count = type_count + 1
+        #if ttls were found in both the nist/ directory and a subdirectory of nist/, mark as TA_1a and have the ta_1a do this check too
+        if ta1_ret_val > -1 and type_count > 1 :        #use > and not == for scaleability? idk dont @ me c:
+            print ("WARNING: Subdirectories found at the same level of .ttl files in the NIST directory.")
+            ta1_ret_val = 0
+        return ta1_ret_val
+    elif dot_count > 1 and dot_count < 4:
+        return dot_count
+    else:
+        eprint("Unkown archive filetype. Please ensure archive name is properly titled per NIST guidelines.")
+        return -1
+
+
+
+def validate_archive_filetype(file):
+    """
+    Checks the input file type. If it is not a valid filetype, the program will
+    print an error message and quit the script.
+    Valid filetypes: .tgz, .tar.gz, or .zip.
+
+    :param file: path to file including file name and extention
+    :return: True if valid filetype and successfully set the global FILETYPE variable, False otherwise
+    :rtype: bool
+    """
+    global FILETYPE
+    file_ext = os.path.splitext(file)[-1].lower()
+    if file_ext == ".gz" or file_ext == ".tgz":
+        FILETYPE = 0
+    elif file_ext == ".zip":
+        FILETYPE = 1
+    else:
+        FILETYPE = -1
+        return False
+
+    return True
+
+def main():
+    """
+
+    :returns: True if script completed successfully, False otherwise
+    :rtype: bool
+    """
+    global FILETYPE
+    global TASK_TYPES
+    global ARCHIVE_NAMES
+
+    #ensure the archive type is a valid filetype and set the global FILETYPE value
+    if not validate_archive_filetype(sys.argv[1]) and FILETYPE >= 0:
+        eprint("FATAL ERROR: Archive filetype unknown. Please use .zip, .tgz, or .tar.gz")
+        exit(False)
+
+    #get names of items in the archive
+    if '-v' in sys.argv: print ("Gathering archive items...")
+    archive = zipfile.ZipFile(sys.argv[1], 'r', allowZip64=True) if FILETYPE else tarfile.open(sys.argv[1], mode='r')
```

Comment:
I did the filetype thing in case there was an additional archive library that was needed to read the archive. However, because python does not support if (int):  as I intended, that would work just as well.